src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -170,15 +149,31 @@ void AsyncLogWriter::write() {
 
 void AsyncLogWriter::run() {
   while (true) {
+    ResourceMark rm;
+    AsyncLogMap<AnyObj::RESOURCE_AREA> snapshot;
     {
       AsyncLogLocker locker;
 
       while (!_data_available) {
         _lock.wait(0/* no timeout */);
       }
+      // Only doing a swap and statistics under the lock to
```

Comment:
@navyxliu 

The original comment was:

>`// lock protection. This guarantees I/O jobs don't block logsites.`

Does this change make sense to you?